From: Jochen Sprickerhof <git@jochen.sprickerhof.de>
Date: Sun, 6 Nov 2016 13:24:12 +0100
Subject: Test for byte order on powerpc

ppc64el uses little endian, relying on gcc to test it.
---
 Foundation/include/Poco/Platform.h | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/Foundation/include/Poco/Platform.h b/Foundation/include/Poco/Platform.h
index def8f07..9387174 100644
--- a/Foundation/include/Poco/Platform.h
+++ b/Foundation/include/Poco/Platform.h
@@ -166,7 +166,11 @@
 #elif defined(__PPC) || defined(__POWERPC__) || defined(__powerpc) || defined(__PPC__) || \
       defined(__powerpc__) || defined(__ppc__) || defined(__ppc) || defined(_ARCH_PPC) || defined(_M_PPC)
 	#define POCO_ARCH POCO_ARCH_PPC
-	#define POCO_ARCH_BIG_ENDIAN 1
+	#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+		#define POCO_ARCH_LITTLE_ENDIAN 1
+	#else
+		#define POCO_ARCH_BIG_ENDIAN 1
+	#endif
 #elif defined(_POWER) || defined(_ARCH_PWR) || defined(_ARCH_PWR2) || defined(_ARCH_PWR3) || \
       defined(_ARCH_PWR4) || defined(__THW_RS6000)
 	#define POCO_ARCH POCO_ARCH_POWER
